src/backend/joanie/core/serializers/admin.py

```diff
@@ -275,22 +299,105 @@ class Meta(AdminOrderGroupSerializer.Meta):
         fields = [*AdminOrderGroupSerializer.Meta.fields, "course_product_relation"]
 
 
-class AdminProductRelationSerializer(serializers.ModelSerializer):
-    """Serializer for CourseProductRelation model."""
+class AdminCourseNestedSerializer(serializers.ModelSerializer):
+    """Serializer for Course model nested in product."""
 
-    organizations = AdminOrganizationLightSerializer(many=True)
-    product = AdminProductSerializer()
-    order_groups = AdminOrderGroupSerializer(many=True, read_only=True)
+    title = serializers.CharField()
+    cover = ThumbnailDetailField(required=False)
+    organizations = AdminOrganizationLightSerializer(many=True, read_only=True)
 
     class Meta:
-        model = models.CourseProductRelation
+        model = models.Course
         fields = (
             "id",
-            "product",
-            "order_groups",
+            "code",
+            "cover",
+            "title",
             "organizations",
+            "state",
         )
-        read_only_fields = ["id", "order_groups"]
+        read_only_fields = ["id", "state"]
+
+
+class AdminCourseProductRelationsSerializer(serializers.ModelSerializer):
+    """
+    Serialize all information about a course relation nested in a product.
+    """
+
+    course = AdminCourseNestedSerializer(read_only=True)
+    product = AdminProductSerializer(read_only=True)
+    organizations = AdminOrganizationLightSerializer(many=True, read_only=True)
+    order_groups = AdminOrderGroupSerializer(many=True, read_only=True)
+
+    class Meta:
+        model = models.CourseProductRelation
+        fields = [
+            "id",
+            "can_edit",
+            "course",
+            "organizations",
+            "order_groups",
+            "product",
+        ]
+        read_only_fields = ["id", "can_edit", "order_groups"]
+
+    def create(self, validated_data):
+        """
+        Create a new course relation and attach provided organizations to it
+        """
+        course_id = self.initial_data["course_id"]
+        course = get_object_or_404(models.Course, id=course_id)
+        validated_data["course"] = course
+
+        product_id = self.initial_data["product_id"]
+        product = get_object_or_404(models.Product, id=product_id)
+        validated_data["product"] = product
+
+        organization_id = self.initial_data.get("organization_id", [])
+        organization = get_object_or_404(models.Organization, id=organization_id)
+        validated_data["organizations"] = [organization]
```

Comment:
You should check if the `initial_data` exists, because if no data is provided, then the `initial_data` property does not exists. Then you should check if the value exists. I think it's safer to check it than thinking that a `null` value will return no result.